lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1566,13 +1566,13 @@ protected function _validateAndCompleteOneToOneMapping(array $mapping)
             $uniqueConstraintColumns = array();
             foreach ($mapping['joinColumns'] as &$joinColumn) {
                 if ($mapping['type'] === self::ONE_TO_ONE && ! $this->isInheritanceTypeSingleTable()) {
-                    if (count($mapping['joinColumns']) == 1) {
-                        if ( ! isset($mapping['id']) || ! $mapping['id']) {
-                            $joinColumn['unique'] = true;
+                        if (count($mapping['joinColumns']) == 1) {
```

Comment:
please fix it. It will be easier to comment on actual changes (we cannot comment directly on the diff with space changes ignored)